doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -727,6 +727,30 @@ declare your constant as rigid for proof search using the command
 Strategies for rewriting
 ------------------------
 
+Usage
+~~~~~
+
+.. tacn:: rewrite_strat @rewstrategy {? in @ident }
+   :name: rewrite_strat
+
+   Rewrite using the strategy s in hypothesis ident or the conclusion.
+
+   .. exn:: Nothing to rewrite.
+
+      If the strategy failed.
+
+   .. exn:: No progress made.
+
+      If the strategy succeeded but made no progress.
```

Comment:
```suggestion
      If the strategy succeeded but made no progress.
```

This is not quite accurate.  Some strategies (`try` and `eval`, maybe some others) are allowed to succeed with no progress, and do not make `rewrite_strat` error.  Other strategies error if there is no progress.

Reply:
This is describing a message which BTW only appears with typeclasses debug.  I think it's fine.

Reply:
Ah, strange, but true